contracts/swap/swap.go

```diff
@@ -105,29 +99,43 @@ func InstanceAt(address common.Address, backend Backend) (Contract, error) {
 }
 
 // Withdraw withdraws amount from the chequebook and blocks until the transaction is mined
-func (s simpleContract) Withdraw(auth *bind.TransactOpts, backend Backend, amount *big.Int) (*types.Receipt, error) {
+func (s simpleContract) Withdraw(auth *bind.TransactOpts, amount *big.Int) (*types.Receipt, error) {
 	tx, err := s.instance.Withdraw(auth, amount)
 	if err != nil {
 		return nil, err
 	}
-	return WaitFunc(auth, backend, tx)
+	return WaitFunc(auth, s.backend, tx)
 }
 
-// Deposit sends a transaction to the chequebook, which deposits the amount set in Auth.Value and blocks until the transaction is mined
-func (s simpleContract) Deposit(auth *bind.TransactOpts, backend Backend, amount *big.Int) (*types.Receipt, error) {
-	rawSimpleSwap := contract.SimpleSwapRaw{Contract: s.instance}
-	if auth.Value != big.NewInt(0) {
-		return nil, fmt.Errorf("Deposit value can only be set via amount parameter")
-	}
-	if amount == big.NewInt(0) {
+// Deposit sends an amount in ERC20 token to the chequebook and blocks until the transaction is mined
+func (s simpleContract) Deposit(auth *bind.TransactOpts, amount *big.Int) (*types.Receipt, error) {
+	if amount.Cmp(big.NewInt(0)) == 0 {
```

Comment:
This is how to do comparisons with *big.Int properly

Reply:
Well yes, but to compare with a zero value:

```
var zero big.Int
if amount.Cmp(&zero) == 0 {
...
```

Reply:
Sorry, what do you mean here? 
I get: `undeclared name: zero`

Reply:
Your proposal does not work. I would welcome a solution to do this in a more elegant way, but have not found one up until now

Reply:
Can you be specific why it does not work?

Reply:
It does work now. Thanks for the elaboration in the private chat